pkg/apis/work/v1alpha1/work_types.go

```diff
@@ -58,6 +58,15 @@ type WorkStatus struct {
 	ManifestStatuses []ManifestStatus `json:"manifestStatuses,omitempty"`
 }
 
+// IsApplied return if the resource is appliied success
+func (w *WorkStatus) IsApplied() bool {
+	if w.Conditions != nil && w.Conditions[0].Type == WorkApplied && w.Conditions[0].Status == metav1.ConditionTrue {
+		return true
+	}
+
+	return false
+}
+
```

Comment:
Maybe we can refactor the function here https://github.com/karmada-io/karmada/blob/master/pkg/controllers/execution/execution_controller.go#L109

Reply:
Not sure if we can extend the `WorkStatus` here.

Reply:
Can you help move this to the `workstatus` controller or to a helper package?

Reply:
@vincent-pli How do you think?

Reply:
Sure

Reply:
Sorry, I didn't make myself clear. I think we can add a common function at [helper](https://github.com/karmada-io/karmada/tree/master/pkg/util/helper) (not extend a `method` for `WorkStatus`).
The API(xxx_types.go) will be referred by third-party systems, they might be don't want it.

Another opinion will be just use [IsStatusConditionTrue](https://github.com/kubernetes/apimachinery/blob/30802a09abe3f7b38aa5749c1ec5f6443a85e6ed/pkg/api/meta/conditions.go#L85)

Reply:
OK, my fault, I used to add these utility function which address the API itself with the API definition. 
I can move it to the `helper`.
BTW, why third-party system do not like this, for the size?
@RainbowMango

Reply:
> BTW, why third-party system do not like this, for the size?

Not really. We need to control the size of `dependencies`. This will help make it easy for people to do some development based on the API. That's the reason why we have [karmada-io/api](https://github.com/karmada-io/api).